wasm-chain-integration/src/utils.rs

```diff
@@ -366,18 +368,31 @@ pub fn get_embedded_schema_v0(bytes: &[u8]) -> ExecResult<schema::ModuleV0> {
     source.get().map_err(|_| anyhow!("Failed parsing schema"))
 }
 
-/// Get the embedded schema if it exists
-pub fn get_embedded_schema_v1(bytes: &[u8]) -> ExecResult<schema::ModuleV1> {
+/// Get the embedded schema if it exists.
+/// It will attempt to use the schema in the custom section "concordium-schema"
+/// and if this is not present it will try to use the custom section
+/// "concordium-schema-v2".
+pub fn get_embedded_schema_versioned(bytes: &[u8]) -> ExecResult<schema::VersionedModuleSchema> {
```

Comment:
The name is a bit misleading as it will also attempt the unversioned. I would just use `get_embedded_schema` and then the doc comments will explain the rest.